baseservice/iodevices/base/input_transaction.py

```diff
@@ -14,18 +12,22 @@ class InputTransaction(metaclass=ABCMeta):
 
     after reading a message from the device (using 'with_transaction=True') the reader should commit/rollback
     the transaction, to signal the device if the message is done processing or not
+
+    calling commit/rollback on a transaction, finishes it. further calls to commit/rollback are ignored
```

Comment:
Question, why do you want to ignore instead of raising an exception? If commit() has been called twice i can see why it is useful (By the end of the function, the message will be committed anyway so the caller will have an expected behavior). However, if the message was already committed when rollback is called, or the message was already rolled back when commit has been called, it would be an unexpected behavior to ignore these situations.